README.md
# confluent.io Cookbook
This cookbook is using the confluent `rpm` and `deb` repositories to install the `confluent-platform` v2 packages. It will create a user `confluent` and init scripts for `kafka` `zookeeper`, `schema-registry` and `kafka-rest`.  

## Recipe
It comes with multiple recipes: `confluent-cookbook::default`, `confluent-cookbook::zookeeper`,`confluent-cookbook::kafka` `confluent-cookbook::kafka-schema-registry` and `confluent-cookbook::kafka-rest`.

## Configuration
Instead of maintaining configuration templates, chef configuration properties will be added as `server.properties`, `zookeeper.properties` , `schema-registry.properties` and `kafka-rest.properties` files.

Example Kafka:
```ruby
default["confluent"]["kafka"]["server.properties"]["broker.id"] = "0"
default["confluent"]["kafka"]["server.properties"]["port"] = "9092"
...
```
will look like this in the `/etc/kafka/server.properties`
```
broker.id=0
port=9092
...
```

Example ZooKeeper:
```ruby
default["confluent"]["kafka"]["zookeeper.properties"]["dataDir"] = "/tmp/zookeeper"
default["confluent"]["kafka"]["zookeeper.properties"]["clientPort"] = "2181"
...
```

will look like this in the `/etc/kafka/zookeeper.properties`
```
dataDir=/tmp/zookeeper
clientPort=2181
...
```

## Other `confluent-platform` configurations
Right now Kafka, ZooKeeper, the Schema-Registry and Kafka-REST can be configured.

## Vagrant
In order to be able to run vagrant you need to run `berks init` and you have to install `vagrant-berkshelf` via `vagrant plugin install vagrant-berkshelf`. Afterwards run `vagrant up`.

## Supported OS Distributions
Right now it has been tested with:
 - CentOS 7.x
 - Ubuntu 14.04 (where it seems to hang sometimes during installation)

## JRE
You need to install your desired JRE yourself, see `Vagrantfile` as an example.

## Build
At some point this project might have a travis build job.


CHANGELOG.md
# 0.1.0

Initial release of confluent-cookbook

# 0.1.1

fixed some renaming

# 0.2.0

Added Schema-Registry service and configuration.

# 0.3.0

Added Kafka REST service and configuration.

# 0.5.0

Updated for Confluent platform 3.0.

# 0.5.1

Converted kafka from SysV to systemd service

attributes/default.rb
default["confluent"]["version"] = "3.0"
default["confluent"]["scala_version"] = "2.11"

# Confluent repository defaults
default["confluent"]["repository"]["rpm"]["url"] = "http://packages.confluent.io/rpm/#{node["confluent"]["version"]}"
default["confluent"]["repository"]["rpm"]["key"] = "http://packages.confluent.io/rpm/#{node["confluent"]["version"]}/archive.key"
default["confluent"]["repository"]["deb"]["url"] = "http://packages.confluent.io/deb/#{node["confluent"]["version"]}"
default["confluent"]["repository"]["deb"]["key"] = "http://packages.confluent.io/deb/#{node["confluent"]["version"]}/archive.key"

# kafka defaults
default["confluent"]["kafka"]["server.properties"]["broker.id"] = "0"
default["confluent"]["kafka"]["server.properties"]["port"] = "9092"
default["confluent"]["kafka"]["server.properties"]["num.network.threads"] = "3"
default["confluent"]["kafka"]["server.properties"]["num.io.threads"] = "8"
default["confluent"]["kafka"]["server.properties"]["socket.send.buffer.bytes"] = "102400"
default["confluent"]["kafka"]["server.properties"]["socket.receive.buffer.bytes"] = "102400"
default["confluent"]["kafka"]["server.properties"]["socket.request.max.bytes"] = "104857600"
default["confluent"]["kafka"]["server.properties"]["log.dirs"] = "/tmp/kafka-logs"
default["confluent"]["kafka"]["server.properties"]["num.partitions"] = "1"
default["confluent"]["kafka"]["server.properties"]["num.recovery.threads.per.data.dir"] = "1"
default["confluent"]["kafka"]["server.properties"]["log.retention.hours"] = "168"
default["confluent"]["kafka"]["server.properties"]["log.segment.bytes"] = "1073741824"
default["confluent"]["kafka"]["server.properties"]["log.retention.check.interval.ms"] = "300000"
default["confluent"]["kafka"]["server.properties"]["log.cleaner.enable"] = "false"
default["confluent"]["kafka"]["server.properties"]["zookeeper.connect"] = "localhost:2181"
default["confluent"]["kafka"]["server.properties"]["zookeeper.connection.timeout.ms"] = "6000"
default["confluent"]["kafka"]["server.properties"]["confluent.support.metrics.enable"] = "false"

# zookeeper defaults
default["confluent"]["kafka"]["zookeeper.properties"]["dataDir"] = "/tmp/zookeeper"
default["confluent"]["kafka"]["zookeeper.properties"]["clientPort"] = "2181"
default["confluent"]["kafka"]["zookeeper.properties"]["maxClientCnxns"] = "0"

# schema registry-defaults
default["confluent"]["kafka"]["schema-registry.properties"]["port"] = "8081"
default["confluent"]["kafka"]["schema-registry.properties"]["kafkastore.connection.url"] = "localhost:2181"
default["confluent"]["kafka"]["schema-registry.properties"]["kafkastore.topic"] = "_schemas"
default["confluent"]["kafka"]["schema-registry.properties"]["debug"] = "false"

# REST proxy defaults
default["confluent"]["kafka"]["kafka-rest.properties"]["id"] = ""
default["confluent"]["kafka"]["kafka-rest.properties"]["schema.registry.url"] = "http://localhost:8081"
default["confluent"]["kafka"]["kafka-rest.properties"]["zookeeper.connect"] = "localhost:2181"
default["confluent"]["kafka"]["kafka-rest.properties"]["consumer.request.max.bytes"] = "67108864"
default["confluent"]["kafka"]["kafka-rest.properties"]["consumer.request.timeout.ms"] = "1000"
default["confluent"]["kafka"]["kafka-rest.properties"]["consumer.threads"] = "1"
default["confluent"]["kafka"]["kafka-rest.properties"]["host.name"] = ""
default["confluent"]["kafka"]["kafka-rest.properties"]["consumer.instance.timeout.ms"] = "300000"
default["confluent"]["kafka"]["kafka-rest.properties"]["consumer.iterator.backoff.ms"] = "50"
default["confluent"]["kafka"]["kafka-rest.properties"]["consumer.iterator.timeout.ms"] = "1"
default["confluent"]["kafka"]["kafka-rest.properties"]["debug"] = "false"
default["confluent"]["kafka"]["kafka-rest.properties"]["metric.reporters"] = ""
default["confluent"]["kafka"]["kafka-rest.properties"]["metrics.jmx.prefix"] = "kafka-rest"
default["confluent"]["kafka"]["kafka-rest.properties"]["metrics.num.samples"] = "2"
default["confluent"]["kafka"]["kafka-rest.properties"]["metrics.sample.window.ms"] = "30000"
default["confluent"]["kafka"]["kafka-rest.properties"]["port"] = "8082"
default["confluent"]["kafka"]["kafka-rest.properties"]["producer.threads"] = "5"
default["confluent"]["kafka"]["kafka-rest.properties"]["request.logger.name"] = "io.confluent.rest-utils.requests"
default["confluent"]["kafka"]["kafka-rest.properties"]["response.mediatype.default"] = "application/vnd.kafka.v1+json"
default["confluent"]["kafka"]["kafka-rest.properties"]["response.mediatype.preferred"] = "[application/vnd.kafka.v1+json, application/vnd.kafka+json, application/json]"
default["confluent"]["kafka"]["kafka-rest.properties"]["shutdown.graceful.ms"] = "1000"
